src/cpp/include/openvino/genai/streamer_base.hpp

```diff
@@ -18,6 +18,7 @@ class OPENVINO_GENAI_EXPORTS StreamerBase {
     /// @brief put is called every time new token is decoded,
     /// @return bool flag to indicate whether generation should be stopped, if return true generation stops
     virtual bool put(int64_t token) = 0;
+    virtual bool put(const std::vector<int64_t>& tokens) = 0;
```

Comment:
That adds a requirements to the child classes to override one more method. I think it must have a default implementation.
`IterableStreamer` only overrides single token version. https://github.com/openvinotoolkit/openvino.genai/blob/9575602e0cf5f3d60452ff711d4a647ba725de33/samples/python/text_generation/multinomial_causal_lm.py#L11 While it passes because bindings handle this, this seems to be correct to follow C++ API in python as well and override both versions if `bool put(const std::vector<int64_t>& tokens)` remains 0 by default

Reply:
> default implementation will be suboptimal as it does not populate `m_tokens_cache` , which is a field of derived class.

A NotIimplementedException is enough.

> Do we need to have `m_tokens_cache` in interface as well?
It will tell users "something" about possible implementation

No, a child class will end up with this field which may not be used. `end()` already suggests that caching is possible.

Reply:
> A NotIimplementedException is enough.

In this case such streamer cannot work with Spec Decoding, Prompt Look-up or even stop strings

It's not BW compatible as well, as in current PR we assume that such method `put(vector)` is availble

Reply:
I see. Then the need to override `IterableStreamer.put()` is even more important.

Reply:
is the final decision is to add default sub-optimal implementation of `put(vector)`? While our streamer will override this method to have optimal implementation

Reply:
Yes. Maybe with a warning print